curriculum/challenges/english/25-front-end-development/lecture-working-with-functions/672d269da46786225e3fe3fd.md

```diff
@@ -10,6 +10,95 @@ dashedName: what-is-the-purpose-of-functions-and-how-do-they-work
 
 Watch the video lecture and answer the questions below.
 
+# --transcript--
+
+What is the purpose of functions, and how do they work?
+
+Functions are reusable pieces of code that perform a specific task or calculate a value. Think of functions as a machine that takes some input, does some operations on it, and then produces an output. Here is an example of declaring a function:
+
+```js
+function greet() {
+  console.log("Hello, Jessica!");
+}
+```
+
+In this example, we have declared a function called `greet`.  Inside that function, we have a `console.log` that logs the message `Hello, Jessica!`. If we tried to run this code, we would not see the message appear in the console. This because we need to call the function.
+
+A function call, or or invocation, is when we actually use or execute the function. To call a function, you will need to reference the function name followed by a set of parentheses:
+
+```js
+function greet() {
+  console.log("Hello, Jessica!");
+}
+
+greet(); // "Hello, Jessica!"
+```
+
+Now the message of `Hello, Jessica!` will be logged to the console. But what if we wanted the message to say `Hello, Nick!` or `Hello, Anna!`? We don’t want to write a new function each time we greet a different user. Instead, we can create a reusable function that uses function parameters and arguments. 
+
+Parameters act as placeholders for the values that will be passed to the function when it is called. They allow functions to accept input and work with that input. Arguments are the actual values passed to the function when it is called. Here is an updated version of the `greet` function that uses parameters and arguments:
+
+```js
+function greet(name) {
+  console.log("Hello, " + name + "!");
+}
+
+greet("Alice"); // Hello, Alice!
+greet("Nick"); // Hello, Nick!
+```
+
+The name serves as the parameter while the strings `Alice` and `Nick` serve as the arguments. Now we have a reusable function that can be used dozens of times throughout our code with different arguments. 
+
+When a function finishes its execution, it will always return a value. By default, the return value will be undefined. Here is an example: 
```

Comment:
```suggestion
When a function finishes its execution, it will always return a value. By default, the return value will be `undefined`. Here is an example: 
```